game/src/mainmenu.js

```diff
@@ -0,0 +1,126 @@
+         */
+        deviceSourceManager.onDeviceConnectedObservable.add((deviceSource) => {
+            // If Mouse/Touch, add an Observer to change text
+            if (deviceSource.deviceType === BABYLON.DeviceType.Mouse || deviceSource.deviceType === BABYLON.DeviceType.Touch) {
+                deviceSource.onInputChangedObservable.add((eventData) => {
+                    if (eventData.type === 'pointerdown' &&
+                        eventData.inputIndex === BABYLON.PointerInput.LeftClick) {
+                        this.onStartGame();
+                    }
+                });
+            }
+            // If Keyboard, add an Observer to change text
+            else if (deviceSource.deviceType === BABYLON.DeviceType.Keyboard) {
+                deviceSource.onInputChangedObservable.add((eventData) => {
+                    if (eventData.type === 'keydown' &&
+                        eventData.key === ' ') {
+                        this.onStartGame();
+                    }
+                });
+            }
+        });
+
+        // This callback is invoked when a new controller is attached:
+        gamepadManager.onGamepadConnectedObservable.add((gamepad, state)=>{
+        
+            // When a new controller is connected add support for detecting button presses
+            gamepad.onButtonDownObservable.add((button, state)=>{
+                this.onStartGame();
+            });
+        });
+    }
+
+    createMainMenu() {
+        this.hudTexture = BABYLON.GUI.AdvancedDynamicTexture.CreateFullscreenUI("Menu");
+    }
+
+    getGreeting() {
+        // Pick a random greeting from the list in constants.js
+        let val = Math.floor(Math.random() * greetings.length);
+        return greetings[val];
+    }
+
+    showUI() {
+        let greeting = this.getGreeting();
+
+        // Create a Text Block that can display the current score
+        this.welcomeText = new BABYLON.GUI.TextBlock();
+        this.welcomeText.text = "Flying Cube Game!";
+        this.welcomeText.fontFamily = "Impact";
+        this.welcomeText.color = "white";
+        this.welcomeText.fontSize = 72;
+        this.welcomeText.verticalAlignment = BABYLON.GUI.TextBlock.VERTICAL_ALIGNMENT_TOP;
+        this.welcomeText.horizontalAlignment = BABYLON.GUI.TextBlock.HORIZONTAL_ALIGNMENT_CENTER;
+        this.welcomeText.width = .5;
+        this.welcomeText.height = .6;
+        
+        this.greetingText = new BABYLON.GUI.TextBlock();
+        this.greetingText.text = greeting;
+        this.greetingText.fontFamily = "Impact";
+        this.greetingText.color = "white";
+        this.greetingText.fontSize = 28;
+        this.greetingText.verticalAlignment = BABYLON.GUI.TextBlock.VERTICAL_ALIGNMENT_TOP;
+        this.greetingText.horizontalAlignment = BABYLON.GUI.TextBlock.HORIZONTAL_ALIGNMENT_CENTER;
+        this.greetingText.width = .5;
+        this.greetingText.height = .7;
+    
```

Comment:
Can you add an instruction on what to do here - like press any key to start, something like that.

Reply:
Should be done now, I've added a "press any button to play" text. You still need to press "A" or "Space" to start though.